gno.land/pkg/sdk/vm/keeper.go

```diff
@@ -73,20 +81,38 @@ func NewVMKeeper(
 	return vmk
 }
 
-func (vm *VMKeeper) Initialize(ms store.MultiStore) {
+func (vm *VMKeeper) Initialize(
+	logger *slog.Logger,
+	ms store.MultiStore,
+	cacheStdlibLoad bool,
+) {
 	if vm.gnoStore != nil {
 		panic("should not happen")
 	}
-	alloc := gno.NewAllocator(maxAllocTx)
 	baseSDKStore := ms.GetStore(vm.baseKey)
 	iavlSDKStore := ms.GetStore(vm.iavlKey)
+
+	if cacheStdlibLoad {
+		vm.gnoStore = CachedStdlibLoad(vm.stdlibsDir, baseSDKStore, iavlSDKStore)
+		return
+	}
```

Comment:
It was hard for me to read the first time, because I was thinking that you were just wrapping, but actually it’s just “setupA” OR “setupB”.

I suggest you create another helper for the noncached gnoStore creation and then you just make an if/else without breaking return.